app/facades/obituary_search.rb

```diff
@@ -21,13 +21,52 @@ def recent_obituaries
     end
   end
 
-  def search_advanced_obituaries(name, year)
-    json = ObituaryService.new.get_advanced_obituaries(name, year)
+  def advanced_query(params)
+    params = parse_params(params)
+    if params.keys.include?(:name) && params.keys.include?(:year)
+      json = ObituaryService.new.advanced_search_name_and_year(params[:name], params[:year])
+    elsif params.keys.include?(:name) || params.keys.include?(:year)
+      if params[:name]
+        json = ObituaryService.new.get_obituaries_by_name(params[:name])
+      elsif params[:year]
+        json = ObituaryService.new.advanced_search_year(params[:year])
+      end
+    end
+    if json.present?
+      json.map do |obituary_data|
+        ApiObituary.new(obituary_data)
+      end
+    end
+  end
 
-    json.map do |obituary_data|
-      ApiObituary.new(obituary_data)
+  def advanced_query_local(params)
+    params = parse_params(params)
+    query_results = []
+    if params.keys.include?(:name)
+      query_results << Obituary.where("first_name ILIKE ? OR last_name ILIKE ?", "%#{params[:name]}%", "%#{params[:name]}%").pluck(:id)
+    elsif params.keys.include?(:location)
+      query_results << Obituary.where("city ILIKE ? OR state ILIKE ?", "%#{params[:location]}%", "%#{params[:location]}%").pluck(:id)
+    elsif params.keys.include?(:age)
+      query_results << Obituary.where("age = ?", "#{params[:age]}").pluck(:id)
+    elsif params.keys.include?(:year)
+      query_results << Obituary.where("extract(year from created_at) = ?", params[:year]).pluck(:id)
     end
-  
+    query_results.flatten.uniq
   end
```

Comment:
Another big dude of a method! Takes into account each possible search param passed, and finds all matches in the database.

Reply:
Awesome, nice use of "ILIKE"